apps/betterangels-backend/hmis/tests/test_mutations.py

```diff
@@ -42,134 +41,216 @@
+        hmis_note = baker.make(
+            HmisNote,
+            hmis_id="480",
+            client_hmis_id="388",
+            hmis_client_profile_id=self.hmis_client_profile.pk,
+            title="prog note title",
+            note="prog note note",
+            date="2011-11-11",
+            created_by=self.org_1_case_manager_1,
+        )
+
+        variables = {
+            "id": str(hmis_note.pk),
+            "hmisId": "480",
+            "clientHmisId": "389",
+            "hmisClientProfileId": str(self.hmis_client_profile.pk),
+            "title": "title",
+            "note": "note",
+            "date": "2025-11-13",
+        }
+        response = self._update_hmis_note_fixture(variables)
+        messages = response["data"]["updateHmisNote"]["messages"]
+
+        self.assertEqual(len(messages), 1)
+        self.assertEqual(
+            messages[0],
+            {
+                "kind": "VALIDATION",
+                "field": None,
+                "message": "Client ID mismatch",
+            },
+        )
+
+    def test_update_hmis_note_mutation_note_id_mismatch(self) -> None:
+        hmis_note = baker.make(
+            HmisNote,
+            hmis_id="480",
+            client_hmis_id="388",
+            hmis_client_profile_id=self.hmis_client_profile.pk,
+            title="prog note title",
+            note="prog note note",
+            date="2011-11-11",
+            created_by=self.org_1_case_manager_1,
+        )
+
+        variables = {
+            "id": str(hmis_note.pk),
+            "hmisId": "481",
+            "clientHmisId": "389",
+            "hmisClientProfileId": str(self.hmis_client_profile.pk),
+            "title": "title",
+            "note": "note",
+            "date": "2025-11-13",
+        }
+        response = self._update_hmis_note_fixture(variables)
+        messages = response["data"]["updateHmisNote"]["messages"]
+
+        self.assertEqual(len(messages), 1)
+        self.assertEqual(
+            messages[0],
+            {
+                "kind": "VALIDATION",
+                "field": None,
+                "message": "Note ID mismatch",
+            },
+        )
 
 
 @override_settings(HMIS_REST_URL="https://example.com", HMIS_HOST="example.com")
-class HmisClientProfileQueryTests(HmisClientProfileBaseTestCase):
+class HmisClientProfileMutationTests(HmisClientProfileBaseTestCase):
     def setUp(self) -> None:
         super().setUp()
 
         self.graphql_client.force_login(self.org_1_case_manager_1)
         self.residence_geolocation = [-118.2437207, 34.0521723]
 
-        self.hmis_client_profile = HmisClientProfile.objects.create(
+        self.hmis_client_profile = baker.make(
+            HmisClientProfile,
             # ID & Metadata Fields
-            hmis_id="1",
-            personal_id="7e401eed7ee14c36a7641ef44626695c",
-            unique_identifier="69E44770D",
+            hmis_id=self._get_random_id(),
```

Comment:
I don't think you need to do random. I think baker has a function to do this built in.